request-handler/src/main/java/au/org/aodn/aws/wps/operation/ExecuteOperation.java

```diff
@@ -30,22 +31,25 @@ public ExecuteOperation(Execute executeRequest) {
 
 
     @Override
-    public String execute(Properties config) {
+    public String execute() {
 
         //  Config items:
         //      queue names
         //      job name
         //      AWS region
         //      status filename
         //      status location
-        String statusS3BucketName = config.getProperty(STATUS_S3_BUCKET_CONFIG_KEY);
-        String statusFileName = config.getProperty(STATUS_S3_FILENAME_CONFIG_KEY);
-        String jobName = config.getProperty(AWS_BATCH_JOB_NAME_CONFIG_KEY);
-        String jobQueueName = config.getProperty(AWS_BATCH_JOB_QUEUE_NAME_CONFIG_KEY);
-        String awsRegion = config.getProperty(AWS_REGION_CONFIG_KEY);
-        String environmentName = config.getProperty(ENVIRONMENT_NAME_ENV_VARIABLE_NAME);
-
-        LOGGER.info("Configuration: " + config.toString());
+        String statusS3BucketName = System.getenv(STATUS_S3_BUCKET_CONFIG_KEY);
```

Comment:
Just noting here that if we change the mechanism for sourcing config again, we'll have to change all these classes again.   This is a form of shotgun surgery.  It may be better to separate how the config is sourced from the config itself.

Reply:
Yeah, you probably need some Java thing like IConfigLoaderFactoryFactoryBuilderFactory to simplify it a bit.

Reply:
This stuff should be accessed via the WpsConfig class in wps-common.

Reply:
https://projects.haykranen.nl/java/

Reply:
https://gist.github.com/lolzballs/2152bc0f31ee0286b722

Reply:
Yep - the idea was that the WpsConfig reads all the properties + passed a Properties object to the XXXOperation classes.